### PR TITLE
Recommended extensions Update to SF resources only

### PR DESCRIPTION
**Pull request by smaddox-sf**
### What does this PR do?
Added note on the Expanded pack of the extensions, added ESLint and removed Git-related ones to keep it focused on extensions that specific to SF dev

### What issues does this PR fix or reference?
@W-8204552@

### CODE REVIEW

File: docs/_articles/en/getting-started/recommended-extensions.md [hunk 1]
```diff
@@ -19,23 +19,20 @@ Keeping your code consistent and nicely formatted is important in any programmin
 
 > NOTE: Prettier also supports Apex formatting through the [Prettier Apex plugin](https://github.com/dangmai/prettier-plugin-apex).
 
-## XML
+## ESLint
 
-VS Code doesn’t ship with rich XML tools by default. This extension, built by Red Hat, provides language support for for XML documents such as the `-meta.xml` or `package.xml` files in your VS Code project.
+Analyzes your JavaScript code to find issues and helps you fix them. As a part of the LWC extension, Salesforce provides specific ESLint rules and you can also add your own.  These preconfigured rules to help you write high quality code.
```
Comment (ekapner) [inline docs/_articles/en/getting-started/recommended-extensions.md:24]: In second sentence let's move the "to" - These preconfigured rules help you to write high-quality code. (or the "to" could be completely removed if you think that sounds better)

File: docs/_articles/en/getting-started/recommended-extensions.md [hunk 2]
```diff
@@ -19,23 +19,20 @@ Keeping your code consistent and nicely formatted is important in any programmin
...
-<https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml>
+<https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint>
 
-## GitHub Pull Requests
 
-This extension makes working with GitHub even easier.
+## XML
 
-<https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github>
+VS Code doesn’t ship with rich XML tools by default. This extension, built by Red Hat, provides language support for for XML documents such as the `-meta.xml` or `package.xml` files in your VS Code project.
```
Comment (ekapner) [inline docs/_articles/en/getting-started/recommended-extensions.md:31]: with GitLens, we're providing the link. Do you not want to do that with the XML tool? And is there an official name for it?

**Review by ekapner — COMMENTED** [review]
Just some minor edits. And deleted my question about link to XML tool; see it now.

File: docs/_articles/en/getting-started/recommended-extensions.md (hunk 1 above)
Comment (smaddox-sf) [reply docs/_articles/en/getting-started/recommended-extensions.md]: Thank you, Emily!  Updates made

**Review by ekapner — APPROVED** [review]

**Review by lcampos — APPROVED** [review]

**Pull request merged**